python/tvm/tensor.py

```diff
@@ -31,8 +31,74 @@ def dtype(self):
 
 itervar_cls = None
 
+
+class TensorOp(_expr.ExprOp):
+    """Overloading binary operators provided by topi.
+
+    When topi is imported, _make_bop is implemented.
+    If _make_bop raises NotImplementedError, the operator is not
+    overloaded (i.e. stick to ExprOp); otherwise, the overloaded
+    operator is applied, as implemented in topi.
+    """
+
+    __op_priority__ = 2
+
+    @staticmethod
+    def _make_bop(lhs, rhs, op):
```

Comment:
Let us not dispatch according to op in str field, instead dispatch to global functions following the same convention as numpy (like tvm.generic.add, multiply), which overloads both for Expr and Tensor